main.py

```diff
@@ -94,8 +95,10 @@ def loop(self):
         while True:
             frame = self.display.get_frame()
             if frame is None:
-                print(colored("Couldn't read from camera", 'red'))
-                break
+                if not self.printed:
+                    print(colored("Couldn't read from camera", 'red'))
+                self.printed = True

```

Comment:
This should be indented.
```suggestion
                    self.printed = True
```